server/src/controllers/client_timeline_entries/getAllClientTimelineEntries.js

```diff
@@ -0,0 +1,49 @@
+const logger = require("pino")();
+const { Op } = require("sequelize");
+const ClientTimelineEntry = require("../../models/client_timeline_entry.model");
+
+const getAllClientTimelineEntriesRequestHandler = async (req, res) => {
+  try {
+    const page = req?.query?.page ? parseInt(req.query.page, 10) : null;
```

Comment:
Hi, just to confirm "Consider this abstracted list of timeline entries [update 4, update 3, update 2, update 1] -- page 1 is currently [update 4 and update 3] for page size 2. Imagine another update happens creating a new timeline entry. Now the list would be [update 5, update 4, update 3, update 2, update 1]. When a tces employee clicks next page now they will see [update 3, update 2], update being on both pages." would be the expected behaviour right? If so, this should be the current behaviour. Was wondering if I can focus on this in the timeline routing where I will interact more with the pagination from the frontend side.

Reply:
Hey Kevin, I guess my explanation was a bit bad since the timeline doesn't exactly follow a page model the same way a table may. The point is if we have something like infinite scrolling to populate the timeline, we don't want new entries being added causing other entries showing up twice. Using the previous example, the timeline (on the frontend) at the beginning may have entries [update 4, update 3]. Then a new entry is added called "update 5". After the user scrolls down in order to see more timeline entires, [update 4, update 3, update 2, update 1] should be visible in the timeline. If we use a standard pagination model without timestamps, accessing those additional results using page 2 of the getAllObjectTimelineEntries request at this point would be [update 3, update 2] since update 5 was added. This would cause the timeline to have entries [update 4, update 3, update 3, update 2] if you just append the results from page 2 of getAllObjectTimelineEntries to the existing list of timeline entries on the frontend.

Of course though I'll leave it up to you to decide how you want to achieve the intended behaviour

Reply:
Oh I see makes sense, yeah I think this will be dependent on the how the frontend is implemented. Because for the timeline entries table, to solve the issue that you talked about above I was not thinking of doing paging, I was thinking we can just fetch the size of the time table so far, so if the user keeps scrolling down, then we increment the amount of entries we want further. For example lets say we are initially at 10 entries showing, we scroll down, then we request 20 entries, keep scrolling down we keep 30 entries, this eensures that every time the user scrolls down, we get the most up to date most recent  x entries.

Reply:
Ok that makes sense, the approach definitely works I just don't know how smooth scrolling will be with it since we are essentially building the whole timeline from scratch whenever a user scrolls down

Reply:
@DanielDervishi I was thinking when the user scrolls to the buttom, we prompt a "Load More" button within the timeline to load 10 more.